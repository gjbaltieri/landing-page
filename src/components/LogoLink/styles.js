import Styled, { css } from 'styled-components';
export const Container = Styled.a`
${({ theme }) => css`
  display: flex;
  font-size: ${theme.font.size.medium};
  text-decoration: none;
  color: inherit;
  > img {
    max-height: 2rem;
  }
`}`;
