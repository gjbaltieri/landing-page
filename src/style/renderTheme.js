import { render } from '@testing-library/react';
import { GlobalStyle } from './globalstyle';
import { theme } from './theme';
import { ThemeProvider } from 'styled-components';
export default function renderTheme(children) {
  return render(
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyle />
    </ThemeProvider>
  );
}
