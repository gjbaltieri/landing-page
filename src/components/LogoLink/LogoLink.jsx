import P from 'prop-types';
import * as Styled from './styles';
import Heading from '../Header/Heading';

export default function LogoLink({ imgSrc = '', text, link }) {
  return (
    <Heading fontSize='base' uppercase>
      <Styled.Container href={link}>
        {!!imgSrc && <img src={imgSrc} alt={text} />}
        {!imgSrc && text}
      </Styled.Container>
    </Heading>
  );
}

LogoLink.propTypes = {
  link: P.string.isRequired,
  text: P.string.isRequired,
  imgSrc: P.string,
};
