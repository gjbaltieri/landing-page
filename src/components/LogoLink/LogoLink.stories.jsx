import React from 'react';
import LogoLink from './LogoLink';

export default {
  title: 'LogoLink',
  component: LogoLink,
  argTypes: {
    imgSrc: { type: 'string' },
    text: { type: 'string' },
    link: { type: 'string' },
  },
};

export const ImageOnly = (args) => (
  <div>
    <LogoLink {...args} />
  </div>
);
ImageOnly.args = {
  imgSrc:
    'https://i2.wp.com/www.verifiedmom.com/wp-content/uploads/2014/07/Netflix_Logo_Print_OneColorPMS.png?fit=8685%2C2333&ssl=1',
  link: 'http://google.com',
  text: 'netflix',
};
export const TextOnly = (args) => (
  <div>
    <LogoLink {...args} />
  </div>
);
TextOnly.args = {
  link: 'http://google.com',
  text: 'netflix',
};
