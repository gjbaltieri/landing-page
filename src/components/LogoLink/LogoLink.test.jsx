import 'jest-styled-components';
import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import renderTheme from '../../style/renderTheme';
import LogoLink from './LogoLink';

describe('<LogoLink/>', () => {
  test('should render LogoLink with correct image', () => {
    renderTheme(<LogoLink link='#target' imgSrc='image.jpg' text='any_text' />);
    expect(screen.getByAltText('any_text')).toHaveAttribute('src', 'image.jpg');
  });
  test('should render LogoLink with correct href', () => {
    renderTheme(<LogoLink link='#target' text='any_text' />);
    expect(screen.getByRole('link', { name: 'any_text' })).toHaveAttribute(
      'href',
      '#target'
    );
  });

  test('should render LogoLink with correct snapshot', () => {
    const { container } = renderTheme(
      <LogoLink link='#target' text='any_text' imgSrc='image.jpg' />
    );
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot();
  });
});
