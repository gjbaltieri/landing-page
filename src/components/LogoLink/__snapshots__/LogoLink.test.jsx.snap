// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink/> should render LogoLink with correct snapshot 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c1 > img {
  max-height: 2rem;
}

.c0 {
  color: #282828;
  font-size: 1.6rem;
  font-weight: 900;
  text-transform: uppercase;
}

@media (max-width:768px) {
  .c0 {
    font-size: 4.0rem;
  }
}

<h1
  class="c0"
  font-size="base"
>
  <a
    class="c1"
    href="#target"
  >
    <img
      alt="any_text"
      src="image.jpg"
    />
  </a>
</h1>
`;
